go/posix-router/main.go

```diff
@@ -40,6 +45,60 @@ import (
 
 var globalCfg config.Config
 
+type rateLimitingServer struct {
+	colgrpc.UnimplementedRateLimiterServiceServer
+	dp *router.DataPlane
+}
+
+func (r *rateLimitingServer) AddRateLimit(ctx context.Context, req *colpb.AddRateLimitRequest) (*colpb.Success, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	r.dp.SyncRateLimiter.Ratelimiter.AddRatelimit(req.Identifier, req.Rate, req.Cbs, time.Now())
+	return &colpb.Success{}, nil
+}
+
+func (r *rateLimitingServer) GetBurstSize(ctx context.Context, req *colpb.GetterForIdentifierRequest) (*colpb.GetBurstSizeResponse, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	cbs, err := r.dp.SyncRateLimiter.Ratelimiter.GetBurstSize(req.Identifier)
+	return &colpb.GetBurstSizeResponse{Cbs: cbs}, err
+}
+
+func (r *rateLimitingServer) GetBurstSizeAndRate(ctx context.Context, req *colpb.GetterForIdentifierRequest) (*colpb.GetBurstSizeAndRateResponse, error) {
+	r.dp.SyncRateLimiter.Lock()
```

Comment:
If we expect this function to be called often, we should consider using a read-write lock.